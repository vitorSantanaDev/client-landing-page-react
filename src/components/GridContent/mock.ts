export default {
  title: 'Title text example',
  content: `<p>Lorem ipsum dolor sit amet, <a href="#">consectetur adipiscing</a> elit. Sed ac lacus pretium urna malesuada dignissim. Sed risus tortor, sagittis eget sem ut, porta imperdiet magna. In fringilla nisi enim. Aenean mollis vehicula nulla, non vulputate eros pretium eget. Vivamus tincidunt accumsan massa non convallis. Aliquam molestie efficitur faucibus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; Vestibulum suscipit convallis mi, vitae ornare dolor hendrerit quis. Nam gravida orci posuere nisi imperdiet posuere. Phasellus quis blandit turpis, eu feugiat velit. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Integer ac malesuada quam. Aliquam eu finibus massa.</p>

  <p>Donec pulvinar volutpat ante quis porttitor. In hac habitasse platea dictumst. Sed eu placerat turpis, vitae pulvinar odio. Pellentesque viverra posuere enim, eu dapibus quam congue sit amet. Sed iaculis eleifend scelerisque. Curabitur ut elit at sem cursus euismod. Praesent ornare, urna vitae sollicitudin tincidunt, sem odio accumsan velit, quis iaculis neque elit vel ante. Suspendisse rhoncus felis at arcu blandit facilisis. Cras euismod varius nibh, a cursus nisi. Integer laoreet posuere sem vel pulvinar. Etiam vel blandit libero. Praesent consectetur tellus et justo lobortis rhoncus.</p>
  `,
  background: false
}
