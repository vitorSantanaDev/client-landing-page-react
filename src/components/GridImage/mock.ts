export default {
  background: false,
  title: 'My grid',
  description:
    'Atque doloribus nobis laudantium esse ut, non commodi maxime distinctio veritatis unde',
  grid: [
    {
      image: {
        altText: 'Teste 1',
        srcImage: 'https://source.unsplash.com/random/800x800?r=1'
      }
    },
    {
      image: {
        altText: 'Teste 2',
        srcImage: 'https://source.unsplash.com/random/800x800?r=2'
      }
    },
    {
      image: {
        altText: 'Teste 3',
        srcImage: 'https://source.unsplash.com/random/800x800?r=3'
      }
    },
    {
      image: {
        altText: 'Teste 4',
        srcImage: 'https://source.unsplash.com/random/800x800?r=4'
      }
    },
    {
      image: {
        altText: 'Teste 5',
        srcImage: 'https://source.unsplash.com/random/800x800?r=5'
      }
    },
    {
      image: {
        altText: 'Teste 6',
        srcImage: 'https://source.unsplash.com/random/800x800?r=6'
      }
    }
  ]
}
