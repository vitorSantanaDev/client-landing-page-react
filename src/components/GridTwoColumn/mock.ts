export default {
  title: 'Grid two column',
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Ad vero, accusantium eum exercitationem in ex quo veniam sed totam, quos ipsam at. Facilis a labore alias sed, natus dolor temporibus?',
  srcImage: 'assets/images/javascript.svg'
}
