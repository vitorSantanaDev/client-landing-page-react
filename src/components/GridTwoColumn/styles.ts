import styled, { css } from 'styled-components'
import { Wrapper as Heading } from '../Heading/styles'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: grid;
    align-items: center;
    gap: ${theme.spacings.large};
    grid-template-columns: 1fr 2fr;

    @media ${theme.media.lteMedium} {
      text-align: center;
      grid-template-columns: 1fr;
    }
    ${Heading} {
      margin-bottom: ${theme.spacings.xlarge};
    }
  `}
`
export const TextContainer = styled.div`
  ${({ theme }) => css`
    @media ${theme.media.lteMedium} {
      margin-bottom: ${theme.spacings.large};
    }
  `}
`
export const ImageContainer = styled.div`
  ${() => css``}
`
export const Image = styled.img`
  ${() => css``}
`
