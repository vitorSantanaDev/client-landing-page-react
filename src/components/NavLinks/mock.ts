export default [
  { children: 'First Link', link: '#target_1', newTab: false },
  { children: 'Second Link', link: '#target_2', newTab: true },
  { children: 'Tertiary Link', link: '#target_3', newTab: false },
  { children: 'Last Link', link: '#target_4', newTab: true },
  { children: 'Test Link', link: '#target_5', newTab: false },
  { children: 'Contact Link', link: '#target_6', newTab: false },
  { children: 'About Link', link: '#target_7', newTab: true }
]
