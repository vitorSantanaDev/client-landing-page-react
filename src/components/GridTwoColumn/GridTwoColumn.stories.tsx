import React from 'react'
import { ComponentStory, ComponentMeta } from '@storybook/react'
import GridTwoColumn from '.'
import mock from './mock'

export default {
  title: 'GridTwoColumn',
  component: GridTwoColumn,
  args: { ...mock }
} as ComponentMeta<typeof GridTwoColumn>

export const Default: ComponentStory<typeof GridTwoColumn> = (args) => (
  <GridTwoColumn {...args} />
)
