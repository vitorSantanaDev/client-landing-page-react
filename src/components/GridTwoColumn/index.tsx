import React from 'react'

import Heading from '../Heading'
import SectionBackground from '../SectionBackground'
import TextComponent from '../TextComponent'
import * as S from './styles'

interface IGridTwoColumnProps {
  title: string
  text: string
  srcImage: string
  background?: boolean
}

const GridTwoColumn: React.FC<IGridTwoColumnProps> = ({
  title,
  text,
  srcImage,
  background
}) => (
  <SectionBackground background={background}>
    <S.Wrapper>
      <S.TextContainer>
        <Heading uppercase colorDark={!background} as="h2">
          {title}
        </Heading>
        <TextComponent>{text}</TextComponent>
      </S.TextContainer>
      <S.ImageContainer>
        <S.Image src={srcImage} alt={title} />
      </S.ImageContainer>
    </S.Wrapper>
  </SectionBackground>
)

export default GridTwoColumn
