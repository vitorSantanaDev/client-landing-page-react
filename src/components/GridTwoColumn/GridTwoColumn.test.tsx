import React from 'react'
import { renderTheme } from '../../styles/theme/render-theme'

import GridTwoColumn from '.'
import mock from './mock'

describe('<GridTwoColumn />', () => {
  it('should render two column grid', () => {
    const { container } = renderTheme(<GridTwoColumn {...mock} />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
