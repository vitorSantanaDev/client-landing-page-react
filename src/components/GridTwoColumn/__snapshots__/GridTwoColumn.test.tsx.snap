// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumn /> should render two column grid 1`] = `
.c5 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c1 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background-color: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 2.4rem;
}

.c2 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
  grid-template-columns: 1fr 2fr;
}

.c2 .c4 {
  margin-bottom: 4.0rem;
}

@media (max-width:768px) {
  .c5 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c2 {
    text-align: center;
    grid-template-columns: 1fr;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <h1
          class="c4 c5"
        >
          Grid two column
        </h1>
        <p
          class="c6"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Ad vero, accusantium eum exercitationem in ex quo veniam sed totam, quos ipsam at. Facilis a labore alias sed, natus dolor temporibus?
        </p>
      </div>
      <div
        class=""
      >
        <img
          alt="Grid two column"
          class=""
          src="assets/images/javascript.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
